[Bug]: Failing go test github.com/smallstep/cli/internal/cast on 32bit patforms

### Steps to Reproduce


Build  step-cli and Eexecute the go tests on linux 32 bit platforms (armhf, armv7 and x86)
The test github.com/smallstep/cli/internal/cast fails with:
           cannot use math.MinInt32 - 1 (untyped int constant -2147483649) as int value in argument to cast.Int32 (overflows)






### Your Environment

* OS linux 32bit  armhf ,  armv7,  x86
* `step` CLI Version -0.28.3

### Expected Behavior

go test is successful on 32bit linux platforms

### Actual Behavior

Failing test  github.com/smallstep/cli/internal/cast_test :

github.com/smallstep/cli/internal/cast_test [github.com/smallstep/cli/internal/cast.test]
internal/cast/cast_test.go:59:40: cannot use math.MinInt32 - 1 (untyped int constant -2147483649) as int value in argument to cast.Int32 (overflows)
internal/cast/cast_test.go:63:40: cannot use math.MaxInt32 + 1 (untyped int constant 2147483648) as int value in argument to cast.Int32 (overflows)
internal/cast/cast_test.go:68:55: cannot use math.MaxUint32 (untyped int constant 4294967295) as int value in argument to cast.Uint32 (overflows)
internal/cast/cast_test.go:77:41: cannot use math.MaxUint32 + 1 (untyped int constant 4294967296) as int value in argument to cast.Uint32 (overflows)
internal/cast/cast_test.go:90:40: cannot use math.MaxUint32 + 1 (untyped int constant 4294967296) as int value in argument to cast.Uint8 (overflows)

see Build Logs:
https://gitlab.alpinelinux.org/wdev/aports/-/jobs/1740037
https://gitlab.alpinelinux.org/wdev/aports/-/jobs/1740032
https://gitlab.alpinelinux.org/wdev/aports/-/jobs/1740031

### Additional Context
the new test was introduced witkh pull request #1363 

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction.
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

--- a/internal/cast/cast_test.go
+++ b/internal/cast/cast_test.go
@@ -36,7 +36,7 @@ func TestInt64ConvertsValues(t *testing.T) {
 }
 
 func TestInt64PanicsOnLargeValue(t *testing.T) {
-	require.Panics(t, func() { cast.Int64(uint64(math.MaxInt + 1)) })
+	require.Panics(t, func() { cast.Int64(uint64(math.MaxInt64 + 1)) })
 }
 
 func TestUint64ConvertsValues(t *testing.T) {
@@ -56,16 +56,16 @@ func TestInt32ConvertsValues(t *testing.T) {
 }
 
 func TestInt32PanicsOnTooSmallValue(t *testing.T) {
-	require.Panics(t, func() { cast.Int32(math.MinInt32 - 1) })
+	require.Panics(t, func() { cast.Int32(int64(math.MinInt32 - 1)) })
 }
 
 func TestInt32PanicsOnLargeValue(t *testing.T) {
-	require.Panics(t, func() { cast.Int32(math.MaxInt32 + 1) })
+	require.Panics(t, func() { cast.Int32(int64(math.MaxInt32 + 1)) })
 }
 
 func TestUint32ConvertsValues(t *testing.T) {
 	require.Equal(t, uint32(0), cast.Uint32(0))
-	require.Equal(t, uint32(math.MaxUint32), cast.Uint32(math.MaxUint32))
+	require.Equal(t, uint32(math.MaxUint32), cast.Uint32(int64(math.MaxUint32)))
 	require.Equal(t, uint32(42), cast.Uint32(42))
 }
 
@@ -74,8 +74,9 @@ func TestUint32PanicsOnNegativeValue(t *testing.T) {
 }
 
 func TestUint32PanicsOnLargeValue(t *testing.T) {
-	require.Panics(t, func() { cast.Uint32(math.MaxUint32 + 1) })
+	require.Panics(t, func() { cast.Uint32(int64(math.MaxUint32 + 1)) })
 }
+
 func TestUint8ConvertsValues(t *testing.T) {
 	require.Equal(t, uint8(0), cast.Uint8(0))
 	require.Equal(t, uint8(math.MaxUint8), cast.Uint8(math.MaxUint8))
@@ -87,5 +88,5 @@ func TestUint8PanicsOnNegativeValue(t *testing.T) {
 }
 
 func TestUint8PanicsOnLargeValue(t *testing.T) {
-	require.Panics(t, func() { cast.Uint8(math.MaxUint32 + 1) })
+	require.Panics(t, func() { cast.Uint8(math.MaxUint8 + 1) })
 }

